python/llm/example/GPU/LLM-Patch-Inference/hf_generate.py

```diff
@@ -0,0 +1,384 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Some parts of this file is adapted from
+# https://github.com/mosaicml/llm-foundry/blob/main/scripts/inference/hf_generate.py 
+# which is licensed under Apache License 2.0
+#
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+import itertools
+import random
+import time
+import warnings
+from argparse import ArgumentParser, ArgumentTypeError, Namespace
+from contextlib import nullcontext
+from typing import Dict, Union
+
+import numpy as np
+import torch
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+from bigdl.llm.llm_patching import llm_patch
+llm_patch()
```

Comment:
Can we put this at the very beginning of the file?

Reply:
the order won't influence the result

Reply:
I mean putting it at the very beginning is more friendly to users and easier to find/read.

Reply:
seems that this function must be called before bigdl import